packages/git/src/node/git-repository-watcher.ts

```diff
@@ -90,18 +90,17 @@ export class GitRepositoryWatcher {
     protected async syncStatus(initial: boolean = false): Promise<boolean> {
         try {
             const source = this.options.repository;
-            const status = await this.git.status(source);
+            const newStatus = await this.git.status(source);
             const oldStatus = this.status;
-            if (initial || !WorkingDirectoryStatus.equals(status, oldStatus)) {
-                this.status = status;
-                this.onStatusChangedEmitter.fire({ source, status, oldStatus });
+            if (initial || !WorkingDirectoryStatus.equals(newStatus, oldStatus)) {
+                this.status = newStatus;
+                this.onStatusChangedEmitter.fire({ source, status: newStatus, oldStatus });
             }
         } catch (error) {
-            // TODO: Is it going to work only with `en` local?
-            if (error.message !== 'Unable to find path to repository on disk.') {
+            if (GitUtils.isRepositoryDoesNotExistError(error)) {
                 return false;
             }
-            // TODO: Do we really want to swallow errors?
+            throw error;
```

Comment:
What do we expect if the error happens? If we want to continue watching then we should only log an error.

Reply:
OK.

Reply:
I assume we want to continue the watching.